node-tests/e2e/test-support-test.js

```diff
@@ -0,0 +1,112 @@
+  setConfig
+} = require('../utils');
+
+describe('e2e: provides test support', function() {
+  this.timeout(300000);
+
+  let app;
+
+  // tests not altering package.json
+  describe('', function() {
+    before(async function() {
+      app = new AddonTestApp();
+
+      await app.create('default', { noFixtures: true });
+
+      // create a simple rendering tests that violates default CSP by using
+      // inline JavaScript
+      let testFolder = 'tests/integration/components';
+      let testFile = `${testFolder}/my-component-test.js`;
+      await fs.ensureDir(app.filePath(testFolder));
+      await fs.writeFile(
+        app.filePath(testFile),
+        "                                                                       \
+          import { module, test } from 'qunit';                                 \
+          import { setupRenderingTest } from 'ember-qunit';                     \
+          import { render } from '@ember/test-helpers';                         \
+          import hbs from 'htmlbars-inline-precompile';                         \
+                                                                                \
+          module('Integration | Component | my-component', function(hooks) {    \
+            setupRenderingTest(hooks);                                          \
+                                                                                \
+            test('it renders', async function(assert) {                         \
+              await render(hbs`<div style='display: none;'></div>`);            \
+              assert.ok(true);                                                  \
+            });                                                                 \
+          });                                                                   \
+        "
+      );
+    });
+
+    afterEach(async function() {
+      await removeConfig(app);
+    });
+
+    it('causes tests to fail on CSP violations', async function() {
+      // runEmberCommand throws result object if command exists with non zero
+      // exit code
+      try {
+        await app.runEmberCommand('test');
+
+        // expect runEmberCommand to throw
+        expect(false).to.be.true;
+      } catch({ code }) {
+        expect(code).to.equal(1);
+      }
+    });
+
+    it('does not cause tests failures if addon is disabled', async function() {
+      await setConfig(app, {
+        enabled: false,
+      });
+      let { code } = await app.runEmberCommand('test');
+
+      expect(code).to.equal(0);
+    });
+
+    it('does not cause tests failures if `failTests` config option is `false`', async function() {
+      await setConfig(app, {
+        failTests: false,
+      });
+      let { code } = await app.runEmberCommand('test');
+
+      expect(code).to.equal(0);
+    });
+  });
+
+  // tests altering package.json
+  describe('', function() {
+    // @ToDo: VersionChecker reports qunit@2.9.2 even so app uses 2.7.1
```

Comment:
Can we implement this using ember-try? I don't think this needs to block landing the change though...

Reply:
I'm not sure how we could use ember-try in the context of an ember-cli-addon-tests apps. The issue here isn't with ember-cli-addon-tests but with ember-cli-version-checker. Haven't had the time yet to open a issue but it seems to make some wrong assumptions leading to wrong version reported.

The test is setting up a correct environment. `npm list` returns the specified version of ember-qunit but ember-cli-version-checker seems to pick up the version used by the host project. It seems to be confused by some internals used by ember-cli-addon-tests. At least that's what I got from a quick investigation.

Will try to find the time for opening a bug report including a reproduction.

Reply:
👍, sounds good I had forgotten RE: this being an ember-cli-addon-tests type test